File: 김현주/H-Index/H-Index.cpp
#include <string>
#include <vector>
#include <algorithm>

using namespace std;

int solution(vector<int> citations) {
    sort(citations.begin(), citations.end(), greater<>()); //내림차순 정렬
    int H = 0;
    for(int i=0;i<citations.size();i++) {
        int count = i+1; //논문의 개수
        if(count <= citations[i]) { //만약 h번 이상 인용된 논문이 h편 이상인 경우
            H = count; //H-Index가 된다.
        }
        else {
            break;
        }
    }
    return H;
}

File: 김현주/H-Index/H-Index.js
function solution(citations) {
    citations.sort((a,b) => b-a); //내림차순 정렬
    
    let h = 0;

    citations.forEach((citation, index) => {
        let count = index+1; //논문의 개수
        if(count <= citation) { //만약 h번 이상 인용된 논문이 h편 이상인 경우
            h = count; //H-Index가 된다.
        }
        else {
            return false; //내림차순 정렬되어있으므로 나머지는 비교할 필요가 없다.
        }
    })

    return h;
}